src/components/App.js

```diff
@@ -1,18 +1,53 @@
+import React from 'react';
 import '../App.css';
 import Display from './Display';
 import ButtonPanel from './ButtonPanel';
 import Calculate from '../logic/calculate';
 
-function App() {
-  const forLinters = Calculate('=', { total: '5', next: '3', operation: '+' });
-  return (
-    <>
-      <div>{forLinters}</div>
-      <Display />
-      <ButtonPanel />
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      total: null,
+      next: null,
+      operation: null,
+    };
+    this.handleClick = this.handleClick.bind(this);
+  }
 
-    </>
-  );
-}
+  handleClick(name) {
+    const { total, next, operation } = this.state;
+    const data = { total, next, operation };
+    const calculation = Calculate(name, data);
+
+    this.setState({
+      total: calculation.total,
+      next: calculation.next,
+      operation: calculation.operation,
+    });
+  }
+
+  render() {
+    const { total, next, operation } = this.state;
+    let result;
+    if (!operation) {
+      result = total;
+    } else if (operation === '+/-') {
+      if (!next || next === '0') {
+        result = total;
+      } else {
+        result = next;
+      }
+    } else {
+      result = !next ? operation : next;
+    }
 
-export default App;
+    return (
+      <>
+        <Display result={result} />
+        <ButtonPanel clickHandler={this.handleClick} />
+
```

Comment:
Kindly remove white space.

Reply:
If you mean white spaces before closing tags, I have to put them because of linters.
![whiteSpaces](https://user-images.githubusercontent.com/65791349/128719947-36ff866a-f620-49fd-8b24-fa7abb6683c2.png)

Reply:
@1v4n4 Nope, I meant `line 49`. clear now?

Reply:
got it!